src/mscorlib/src/System/Collections/Generic/List.cs

```diff
@@ -291,10 +299,15 @@ public int BinarySearch(T item, IComparer<T> comparer)
 
     
         // Clears the contents of List.
-        public void Clear() {
+        public void Clear()
+        {
             if (_size > 0)
             {
-                Array.Clear(_items, 0, _size); // Don't need to doc this but we clear the elements so that the gc can reclaim the references.
+                if (JitHelpers.ContainsReferences<T>())
```

Comment:
Assuming the call completely evaporates, should we use this in List (and maybe elsewhere in other data structures) anywhere we null out a field, e.g. not just Clear but all of the Remove methods?